includes/Core/Util/Debug_Data.php

```diff
@@ -0,0 +1,272 @@
+				'value' => GOOGLESITEKIT_VERSION,
+			),
+			'php_version'    => array(
+				'label' => __( 'PHP Version', 'google-site-kit' ),
+				'value' => PHP_VERSION,
+			),
+			'wp_version'     => array(
+				'label' => __( 'WordPress Version', 'google-site-kit' ),
+				'value' => get_bloginfo( 'version' ),
+			),
+			'reference_url'  => array(
+				'label' => __( 'Reference Site URL', 'google-site-kit' ),
+				'value' => $this->context->get_reference_site_url(),
+			),
+			'amp_mode'       => $this->get_amp_mode_field(),
+			'site_status'    => $this->get_site_status_field(),
+			'user_status'    => $this->get_user_status_field(),
+			'active_modules' => $this->get_active_modules_field(),
+		);
+		$none   = __( 'None', 'google-site-kit' );
+
+		return array_map(
+			function ( $field ) use ( $none ) {
+				if ( empty( $field['value'] ) ) {
+					$field['value'] = $none;
+					$field['debug'] = 'none';
+				}
+
+				return $field;
+			},
+			array_merge( $fields, $this->get_module_fields() )
+		);
+	}
+
+	/**
+	 * Gets the field definition for the amp_mode field.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	private function get_amp_mode_field() {
+		$mode     = $this->context->get_amp_mode();
+		$mode_map = array(
+			'primary'   => __( 'Primary', 'google-site-kit' ),
+			'secondary' => __( 'Secondary', 'google-site-kit' ),
+		);
+
+		return array(
+			'label' => __( 'AMP Mode', 'google-site-kit' ),
+			'value' => isset( $mode_map[ $mode ] ) ? $mode_map[ $mode ] : __( 'No', 'google-site-kit' ),
+			'debug' => $mode,
+		);
+	}
+
+	/**
+	 * Gets the field definition for the site_status field.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	private function get_site_status_field() {
+		$credentials  = new Credentials( new Encrypted_Options( new Options( $this->context ) ) );
+		$is_connected = $credentials->has();
+		$using_proxy  = ( new Authentication( $this->context ) )->get_oauth_client()->using_proxy();
+		$status_map   = array(
+			'connected-site'  => __( 'Connected through site credentials', 'google-site-kit' ),
+			'connected-oauth' => __( 'Connected through OAuth client credentials', 'google-site-kit' ),
+			'not-connected'   => __( 'Not connected', 'google-site-kit' ),
+		);
+
+		if ( $is_connected && $using_proxy ) {
+			$status = 'connected-site';
+		} elseif ( $is_connected && ! $using_proxy ) {
+			$status = 'connected-oauth';
+		} else {
+			$status = 'not-connected';
+		}
+
+		return array(
+			'label' => __( 'Site Status', 'google-site-kit' ),
+			'value' => $status_map[ $status ],
+			'debug' => $status,
+		);
+	}
+
+	/**
+	 * Gets the field definition for the user_status field.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	private function get_user_status_field() {
+		$user_options      = new User_Options( $this->context );
+		$has_auth_token    = (bool) $user_options->get( OAuth_Client::OPTION_ACCESS_TOKEN );
+		$has_refresh_token = (bool) $user_options->get( OAuth_Client::OPTION_REFRESH_TOKEN );
```

Comment:
These two should instead be using `Authentication::get_oauth_client()` and then its respective `get_access_token()` and `get_refresh_token()` methods, rather than relying on low-level option lookup.

Reply:
Do we need to check for the refresh token specifically? I would think we would want to use the same as `Authentication::is_authenticated`. If we do want to include the refresh token in this logic, should we not do it in that method instead?

Reply:
That's a good point. I think we should just rely on `is_authenticated` for now (technically the user is authenticated with just an access token). However I'd say it's potentially worth expanding the Site Health information specifically in the future with
* whether there is also a refresh token
* whether the access token is already expired